packages/vhd-lib/src/Vhd/VhdAbstract.integ.spec.js

```diff
@@ -139,3 +150,81 @@ test('It create , rename and unlink alias', async () => {
     expect(await fs.exists(aliasFileNameRenamed)).toEqual(false)
   })
 })
+
+test('it can create a vhd stream', async () => {
+  const initalSize = 6
+  const rawFileName = `${tempDir}/randomfile`
+  await createRandomFile(rawFileName, initalSize)
+  const vhdFileName = `${tempDir}/vhd.vhd`
+  await convertFromRawToVhd(rawFileName, vhdFileName)
+  const bat = Buffer.alloc(512)
+
+  await Disposable.use(async function* () {
+    const handler = yield getSyncedHandler({ url: 'file://' + tempDir })
+
+    // mark first block as unused
+    await handler.read('vhd.vhd', bat, 1024 + 512)
+    bat.writeUInt32BE(BLOCK_UNUSED, 0)
+    await handler.write('vhd.vhd', bat, 1024 + 512)
+
+    const vhd = yield openVhd(handler, 'vhd.vhd')
+    await vhd.readBlockAllocationTable()
+    const stream = vhd.stream()
+
+    // read all the stream into a buffer
+
+    const buffer = await streamToBuffer(stream)
+    const length = buffer.length
+    const start = 512 /* footer */ + 1024 /* header */ + 512 /* BAT */
+    const footer = buffer.slice(0, 512)
+    expect(length).toEqual(start + 2 * vhd.fullBlockSize + 512 /* footer again */)
```

Comment:
Would you mind putting the number of blocks in a constant above and then compute derived values?

It will make the code easier to read and change if necessary.